modules/gapi/src/api/render_ocv.cpp

```diff
@@ -16,16 +16,27 @@ namespace draw
 // FIXME Support `decim` mosaic parameter
 inline void mosaic(cv::Mat& mat, const cv::Rect &rect, int cellSz)
 {
-    cv::Mat msc_roi = mat(rect);
+    cv::Rect mat_rect(0, 0, mat.cols, mat.rows);
+    auto intersection = mat_rect & rect;
+
+    cv::Mat msc_roi = mat(intersection);
     int crop_x = msc_roi.cols - msc_roi.cols % cellSz;
     int crop_y = msc_roi.rows - msc_roi.rows % cellSz;
 
+    cv::Mat cell_roi;
     for(int i = 0; i < crop_y; i += cellSz ) {
         for(int j = 0; j < crop_x; j += cellSz) {
-            auto cell_roi = msc_roi(cv::Rect(j, i, cellSz, cellSz));
+            cell_roi = msc_roi(cv::Rect(j, i, cellSz, cellSz));
             cell_roi = cv::mean(cell_roi);
         }
     }
+
+    // Handle tail
+    cell_roi = msc_roi(cv::Rect(crop_x, 0, msc_roi.cols - crop_x, crop_y));
```

Comment:
- Proper handling of "no tail" cases
- Are there two rectangles only for the tail processing?
- I believe you should not use rectangles greater than `cellSz x cellSz`. `0 .. crop_y` violates this condition.

Reply:
Fixed

Reply:
> Proper handling of "no tail" cases

Test still crashes in "Debug" build due extra checks (don't look on pre-commit, it doesn't care about 720p test cases).